scipy/signal/_peak_finding.py

```diff
@@ -246,6 +250,107 @@ def argrelextrema(data, comparator, axis=0, order=1, mode='clip'):
     return np.nonzero(results)
 
 
+def _handle_arg_x(value):
+    """Check and normalize argument `x`.
+
+    See descriptions in functions `find_peaks`, `peak_prominences` and
+    `peak_widths`.
+
+    Parameters
+    ----------
+    value : array_like
+        A one-dimensional array.
+
+    Returns
+    -------
+    value : ndarray
+        A one-dimensional C-contiguous array with dtype('float64').
+    """
+    value = np.asarray(value, order='C', dtype=np.float64)
+    if value.ndim != 1:
+        raise ValueError('`x` must have exactly one dimension')
+    return value
+
+
+def _handle_arg_peaks(value):
+    """Check and normalize argument `peaks`.
+
+    See descriptions in functions `peak_prominences` and `peak_widths`.
+
+    Parameters
+    ----------
+    value : array_like
+        A one-dimensional array.
+
+    Returns
+    -------
+    value : ndarray
+        A one-dimensional C-contiguous array with dtype('intp').
+    """
+    value = np.asarray(value)
+    if value.size == 0:
+        # Empty arrays default to np.float64 but are valid input
+        value = np.array([], dtype=np.intp)
+    try:
+        # Safely convert to C-contiguous array of type np.intp
+        value = value.astype(np.intp, order='C', casting='safe',
+                             subok=False, copy=False)
+    except TypeError:
+        raise TypeError("Cannot safely cast `peaks` to dtype('intp')")
+    if value.ndim != 1:
+        raise ValueError('`peaks` must have exactly one dimension')
+    return value
+
+
+def _handle_arg_wlen(value):
+    """Check and normalize argument `wlen`.
+
+    See descriptions in functions `peak_prominences` and `peak_widths`.
+
+    Parameters
+    ----------
+    value : number or None
+        None or a number which must be larger than 1.
+
+    Returns
+    -------
+    value : int
+        The original `value` rounded up to the next integer or -1 if `value`
+        was None.
+    """
+    if value is None:
+        value = -1  # _peak_prominences expects int -> None == -1
+    elif 1 < value:
+        # Round up to next positive integer; rounding up to next odd integer
+        # happens implicitly inside the inner function
+        value = int(math.ceil(value))
```

Comment:
Just take `int()`, we can't keep track of numerical noise say 1.000000000001

Reply:
This behavior is already a documented part of the AP (see [peak_prominences](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.peak_prominences.html#scipy.signal.peak_prominences)). I think I explicitly wanted to allow for float because it allows for the specification of the window length as a ratio of the sampling rate without worrying about conversion. 

If you deem this change worth a depreciation I can of course change it but I would prefer not to. In retrospect I agree that a simple `int()` may have been better.

Reply:
Hmm, I should have read more about the function before. It is indeed more complicated than it seems. However, hitting a perfect int as a float is always a problematic endeavor. So I think we might have less headache if we force people to give the ints themselves. This is a really tough spot for debugging.

Reply:
I don't like breaking backward compat but it's unlikely that we will do the most unexpected thing in the case of `float` input. I don't mind forcing people to pass int, which is usually done in SciPy with:
```
value = operator.index(value)
```
Let's allow the users to do the conversion and deal with the tradeoffs themselves.

Reply:
> The current status is that we round up wlen to the next odd integer regardless of wlen being an integer or float. This behavior is clearly documented in the arguments description

Okay makes sense. Let's not add any more adjustments, then, and keep the current behavior for `float`.